### PR TITLE
Create installable repo with pytest asdf schema tester

### PR DESCRIPTION
**Pull request by jdavies-st**
In order to make the pytest asdf schema checker work alongside the version supplied with the `asdf` package, we needed to use differently-named command-line switch to turn on the tests.  So instead of 

```
pytest --asdf-tests
```

it is now

```
pytest --asdf-schema
```

And I've removed the `asdf_schema_tests_enabled` config option, which does the same thing.  It is unnecessary as one can simply do the following in `setup.cfg` and get the same behavior:

```
[tool:pytest]
addopts = --asdf-schema
```

Better to have only one way to do something rather than two.  Otherwise it works the same as existing plugin.

### CODE REVIEW

File: pytest_asdf_schema/plugin.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+import pytest
+
+import numpy as np
+
+# Avoid all imports of asdf at this level in order to avoid circular imports
+
+
+def pytest_addoption(parser):
+    parser.addini(
+        "asdf_schema_location", "Root path indicating where schemas are stored")
```
Comment (eslavich) [inline pytest_asdf_schema/plugin.py:18]: Now that I see them up close, I suspect this similar-but-different set of names is going to confuse me.  What do you think about this kind of layout in setup.cfg:

```ini
[asdf_schema_helpers]
schema_root = schemas
```
Then in `plugin.py` we'd need to drop `pytest_addoption` and instead do something like this in our `pytest_collect_file` function:
```python
config = configparser.ConfigParser()
config.read(parent.config.inifile)

if not config.has_section("asdf_schema_helpers"):
  return

schema_root = config["asdf_schema_helpers"].get("schema_root", None)
# etc
```

Comment (eslavich) [reply pytest_asdf_schema/plugin.py:18]: (btw, the setuptools plugin is going to need access to some of these, which is some additional motivation to get them out of `[tool:pytest]` and into another section where it makes sense to share)

File: pytest_asdf_schema/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
```
Comment (eslavich) [inline pytest_asdf_schema/__init__.py:1]: How do you think the `src/` structure worked out with asdf-transform-schemas?  I like it so far.

Comment (jdavies-st) [reply pytest_asdf_schema/__init__.py:1]: I think it is good.  Shall we adapt that structure to this repo as well?

File: pytest_asdf_schema/plugin.py (hunk 1 above)
Comment (jdavies-st) [reply pytest_asdf_schema/plugin.py:18]: Yeah, that makes more sense, though it's a little pytest non-standard.

Comment (jdavies-st) [reply pytest_asdf_schema/plugin.py:18]: And it means that if a do `pytest --help` these config options will not show up the help, as they would if used `pytest_addoption`.  Very non-standard.  Hmm.

Comment (eslavich) [reply pytest_asdf_schema/plugin.py:18]: Oh that's not good, I didn't realize they were listed on the in the help.

It looks like this sort of thing works:

```ini
[tool:pytest]
asdf.schema_root
```

Maybe all the new ones can share a namespace with a dot?

File: pytest_asdf_schema/__init__.py (hunk 2 above)
Comment (eslavich) [reply pytest_asdf_schema/__init__.py:1]: Yes, let's src all the things!

File: pytest_asdf_schema/plugin.py (hunk 1 above)
Comment (jdavies-st) [reply pytest_asdf_schema/plugin.py:18]: Oh nice!  Perfect. Thank you.

File: pytest_asdf_schema/__init__.py (hunk 2 above)
Comment (jdavies-st) [reply pytest_asdf_schema/__init__.py:1]: all-the-things

File: pytest_asdf_schema/plugin.py (hunk 1 above)
Comment (jdavies-st) [reply pytest_asdf_schema/plugin.py]: So it turns out the ini config options can be named the same, but the command line arg for turning on the tests has to be unique.

File: src/pytest_asdf_schema/plugin.py [hunk 3]
```diff
@@ -21,12 +19,6 @@ def pytest_addoption(parser):
     parser.addini(
         "asdf_schema_skip_examples",
         "Base names of schemas whose examples should not be tested")
-    parser.addini(
```
Comment (eslavich) [inline src/pytest_asdf_schema/plugin.py:24]: Is the idea that we can use `addopts` to add `--asdf-schema` when we want to change the default?

Comment (jdavies-st) [reply src/pytest_asdf_schema/plugin.py:24]: Right.  I updated the PR message above to indicate this.  I clearly also need to pull over some documentation on how to use this as well. I'll do that in a separate PR.

**Pull request merged**